### ISSUE TITLE
`diary_frequency` in second wiki is only respected when a file from the wiki is already open

### ISSUE THREAD

**Issue by ferdinandyb** [vimwiki/vimwiki#1365]
Prior to submitting a new issue make sure to complete these steps:

- [x] Include the VimWiki settings from your `.vimrc`
- [x] Include the syntax you are using (default / Markdown / MediaWiki) 
- [x] Provide a detailed description of the problem including **steps to reproduce the issue**.
- [x] Include the output of `:VimwikiShowVersion`.

# Config
Vimwiki config:
```viml
let g:vimwiki_list = [{'path':'~/testwiki'},{'path': '~/org/', 'syntax': 'markdown', 'ext': '.md', 'diary_frequency':'weekly', 'auto_diary_index':1}]
```

This issue is with the second _markdown_ wiki.

```
:VimwikiShowVersion
Version: 2023.05.12
Os: Linux
Vim: 900
Branch: dev
Revision: f0fe154
Date: 2023-07-31 08:27:35 -0400
```

# Issue

Setting `diary_frequency` to `weekly` is only respected when you actually have a file from the corresponding wiki open, otherwise it uses the first wiki's

# To reproduce:

```viml
let g:vimwiki_list = [{'path':'~/testwiki'},{'path': '~/org/', 'syntax': 'markdown', 'ext': '.md', 'diary_frequency':'weekly', 'auto_diary_index':1}]
```

1. use the above config
2. open vim and execute `:VimwikiMakeYesterdayDiary 2` -> observe it literally creates yesterday's date
4. open any file from the second wiki (the previously created diary entry is also good)
5. execute `:VimwikiMakeYesterdayDiary 2`  -> observe that not it correctly creates last week's file

```viml
let g:vimwiki_list = [{'path':'~/testwiki', 'diary_frequency':'monthly'},{'path': '~/org/', 'syntax': 'markdown', 'ext': '.md', 'diary_frequency':'weekly', 'auto_diary_index':1}]
```
6. switch to the above config
7. open vim and execute `:VimwikiMakeYesterdayDiary 2` -> observe it now creates the previous month's diary note

**Comment by ferdinandyb** [comment]
So the problem is, that `vimwiki#diary#diary_date_link` is called, but this function decides the `wiki_nr` for itself [here](https://github.com/vimwiki/vimwiki/blob/f0fe154ede6b11e3db9b058b930005a056a3d1c6/autoload/vimwiki/diary.vim#L41), and uses that to [get the frequency](https://github.com/vimwiki/vimwiki/blob/f0fe154ede6b11e3db9b058b930005a056a3d1c6/autoload/vimwiki/diary.vim#L70).

**Comment by tinmarino** [comment]
Fixed by #1365. Good job @ferdinandyb
